ext/web/stream_resource.rs

```diff
@@ -471,18 +475,36 @@ pub fn op_readable_stream_resource_write_buf(
   }
 }
 
-#[op2(async)]
+/// Write to the channel synchronously, returning 0 if the channel was closed, 1 if we wrote
+/// successfully, 2 if the channel was full and we need to block.
+#[op2]
+pub fn op_readable_stream_resource_write_sync(
+  sender: *const c_void,
+  #[buffer] buffer: JsBuffer,
+) -> u32 {
+  let sender = get_sender(sender);
+  if sender.can_write() {
+    if sender.closed() {
+      0
+    } else {
+      sender.write(buffer.into_parts()).unwrap();
+      1
+    }
+  } else {
+    2
+  }
+}
+
+#[op2(fast)]
 pub fn op_readable_stream_resource_write_error(
   sender: *const c_void,
   #[string] error: String,
-) -> impl Future<Output = bool> {
+) -> bool {
   let sender = get_sender(sender);
-  async move {
-    // We can always write an error, no polling required
-    // TODO(mmastrac): we can remove async from this method
-    sender.write_error(type_error(Cow::Owned(error)));
-    !sender.closed()
-  }
+  // We can always write an error, no polling required
+  // TODO(mmastrac): we can remove async from this method
```

Comment:
I'm not sure what this TODO means, could you elaborate?

Reply:
I forgot to remove this TODO after I fixed this, good catch :)